crates/primitives-traits/src/header/sealed.rs

```diff
@@ -7,29 +7,53 @@ use alloy_rlp::{Decodable, Encodable};
 use bytes::BufMut;
 use core::mem;
 use derive_more::{AsRef, Deref};
-use reth_codecs::add_arbitrary_tests;
+use reth_codecs::{add_arbitrary_tests, Compact};
 use serde::{Deserialize, Serialize};
 
 /// A [`Header`] that is sealed at a precalculated hash, use [`SealedHeader::unseal()`] if you want
 /// to modify header.
 #[derive(Debug, Clone, PartialEq, Eq, Hash, AsRef, Deref, Serialize, Deserialize)]
 #[add_arbitrary_tests(rlp)]
-pub struct SealedHeader {
+pub struct SealedHeader<H = Header> {
```

Comment:
I think that's reasonable, although for header op and eth use the same type

Reply:
we shouldn't only limit this to op

Reply:
makes sense